packages/block-editor/src/store/actions.js

```diff
@@ -874,6 +872,16 @@ export const __unstableSplitSelection =
 			typeof selectionB.attributeKey === 'string'
 				? selectionB.attributeKey
 				: findRichTextAttributeKey( blockBType );
+		const blockAttributes = select.getBlockAttributes(
+			selectionA.clientId
+		);
+		const bindings = blockAttributes?.metadata?.bindings;
+
+		// If the attribute is bound, don't split the selection and insert a new block instead.
+		if ( bindings?.[ attributeKeyA ] ) {
+			dispatch.insertAfterBlock( selectionA.clientId );
```

Comment:
I'd rather have it not do anything tbh

Reply:
There's also some other nuances: what happens when you paste block into a bound attribute right now with this PR? The correct behaviour would be to paste everything inline imo (since it's an isolated editor)

Reply:
> I'd rather have it not do anything tbh

I don't have a strong opinion, to be honest. I think I lean towards adding a new block but I am not sure. Maybe @jasmussen can help to decide. There are basically two options when a block is connected and users press enter:

**1. Add a new block**


https://github.com/WordPress/gutenberg/assets/34552881/db1d9396-5ffd-4fad-96e9-e8fb5a33d8d0


**2. Don't do anything**

https://github.com/WordPress/gutenberg/assets/34552881/468816aa-129b-4d78-b7b6-1fee5be69998

Reply:
> There's also some other nuances: what happens when you paste block into a bound attribute right now with this PR? The correct behaviour would be to paste everything inline imo (since it's an isolated editor)

Right now, if you copy text it works as expected, but if you copy a whole block, it does nothing. I agree that probably the expected behavior is to paste everything inline. I'll try to take a look at that. Any clue where that logic lives?

Reply:
Just making sure I understand: we're discussing what happens when you press Enter in a bound block, is that right?

Assuming I understand this right, I would lean towards pressing Enter making a new paragraph. I.e. not changing the writing flow behavior just for bound blocks.

Reply:
For now, I added a warning in [this commit](https://github.com/WordPress/gutenberg/pull/61753/commits/2012ae84396ffb0d96829830d36483365555a733) when trying to paste a block into another one with bindings.

I believe changing the value of the meta field (or other source) after pasting might be trickier because we might need to call `setValues`, but I'll keep exploring to see if we can avoid that.

Reply:
> Just making sure I understand: we're discussing what happens when you press Enter in a bound block, is that right?

Sorry I missed the comment. Yes, that was my question 🙂

Reply:
Based on the feedback, I kept the logic to insert a new block when pressing Enter and not allowing to paste blocks when bindings exist until we figure out how to do it properly.

Is that okay for you, @ellatrix ? Any more concerns I should work on to get this merged?

Reply:
I can confirm that `shift+enter` will add a line break while maintaining the text edition. 👍

Reply:
If we want to be like actual paragraph, all the content from the cursor to the end of the paragraph should be moved to the next block.